pkg/search/proxy/framework/plugins/cache/scheme.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// cacheScheme is the runtime.Scheme for the cache plugin.
+var cacheScheme = runtime.NewScheme()
+
+func init() {
+	// This is to ensure that the cache plugin can handle the field selectors for these resources.
```

Comment:
Resources that are not listed here can't be processed, can they? So how do you determine which resources can be listed here? In other words, are there any plans to add resources that are not listed?

Reply:
Thanks @XiShanYongYe-Chang I checked the resource kinds of Kubernetes and the corev1 group is the most important and the most common used. IMO, we should support the corev1 resource kinds first, and the other resource types can be gradually added based on community feedback.

Reply:
@SataQiu, thanks for your quick reply!

Where does the Kubernetes community maintain Corev1 resources?

> Waiting for the Kubernetes runtime scheme package to support automatic ConversionFuncs registration, the community is trying to remove the legacyscheme

Is the Kuberenetes community working on this plan? Do we have to rely on this plan to add resources in the future?

Reply:
Let me understand method two again, which means that when the upstream community implements this, other resources will automatically support it, and we don't need to register additional conversion functions. Am I right?

Reply:
Yes, you understand correctly.

Reply:
Thanks~

> According to users feedback, support more resource types, such as apps, these can be added in subsequent PRs

I think we can do what you say. In addition, can you help create an issue to track this task, and if we don't get feedback on other resources for a long time, we may be able to proactively plan in the future.

Reply:
Would it be better if we added a comment to the file stating that the implementation is referenced from xxx?

Reply:
Good idea, updated.